### DISCUSSION TITLE
API & kitten to perform the same function as scrollback_fill_enlarged_window

### DISCUSSION THREAD

**Discussion by Lillecarl** [kovidgoyal/kitty#7086] [Ideas]
Hi!

As you can see on the window, when I'm searching through my command history i get a lot of dead space under the cursor once the command has been selected, which can be quite cumbersome when you're in small windows, as you can see that once you start typing it scrolls back to that same state again.

With scrollback_fill_enlarged_window enabled it seems like this can be cleared by resizing the window(with my window manager), which is great but not the optimal workflow 😄 

I've tried using the screen resize kitten, but it seems like linebuf != main_linebuf when I'm executing (?) so it doesn't trigger [the scrollback edit](https://github.com/kovidgoyal/kitty/blob/master/kitty/screen.c#L432).

It's not just atuin, fish search, fish completions and other things also write things below the prompt which is then entirely useless screen real estate.

I've attached a video showcasing the "issue", i gave an implementation a shot but failed 😸 

https://github.com/kovidgoyal/kitty/assets/207073/3bcfe78d-f9aa-4cac-8d45-d9201f2c910e

kitty 0.32.1 (NixOS ❤️)
fish 3.7.0

Thanks for your great work!

**Comment by kovidgoyal** [comment]
See https://sw.kovidgoyal.net/kitty/unscroll/

**Reply by Lillecarl** [reply]
Hi Kovid!

Thanks for the quick reply. I'm not a Shell Guru™️, but I tried implementing this into the tool I'm using for hist search (atuin). Here's a video showing what I ended up with.

https://github.com/kovidgoyal/kitty/assets/207073/f711ff05-8b8a-42d5-b493-a82c092f000d

In the end I show the source for the fish function I've edited, but the gist is that this runs:
```
    commandline -f repaint # <- this is default from atuin
    echo -e '\e[25+T'
    echo '\\e[25+T sent in function end' >&3
```

That's my interpretation of https://en.wikipedia.org/wiki/ANSI_escape_code#CSI_(Control_Sequence_Introducer)_sequences + https://sw.kovidgoyal.net/kitty/unscroll/ together.

I'm certain I've misunderstood something, I'd appreciate if you could shed some further light on this 😄

**Reply by kovidgoyal** [reply]
I dont use fish so I cant really help you with it. What you need to do is emit that escape sequence before the command prompt is redrawn by the number of lines you want unscrolled. And then get fish to draw the prompt on the line at which the cursor is now positioned which will be N lines after the line it was at before you emitted the escape code. Whether fish provides enough integration points to accomplish this, I have no idea.

And if you are having trouble with the escape code, you can use the kitty remote control facility to do the samething, kitten @scroll-window -h

**Reply by Lillecarl** [reply]
Hi again Kovid!

After more reading, banging my head against the wall and testing random bogus I came up with a solution that works!

fish:
```fish
    # %b interprets escape codes
    printf %b '\e[24+T'
    printf %b '\e[24B'
```
any shell (coreutils)
```bash
    # -e interprets escape codes, -n doesn't print a newline
    echo -en '\e[24+T'
    echo -en '\e[24B'
```
Running this before the fish prompt repaint does the trick for me!

Thanks for the help 😺
